Sources/web3swift/Convenience/Decodable+Extensions.swift

```diff
@@ -8,106 +8,86 @@
 import BigInt
 import Foundation
 
-struct AnyCodingKey: CodingKey {
```

Comment:
Actually yes. I wish to discuss it with you tomorrow in about 13:00 gtm + 3 if you mind.

But in short to implement decoding of struct which comes by `feeHistory` I've to totally rewrite decoding hex extension. I don't mind to bring back all required `decodeIfPrecent` methods, but just in true generic way, which i think would cost low amount of work, but coz you're user of those i want to hear from you the use cases of those.

Reply:
Ah, well, it’s ok to split those changes to external pr. I’ll do that tomorrow, and leave here only gas prediction part itself.

Reply:
Sure we can chat tomorrow, I should be available. 

I don't mind the change, just saw it as a more generic improvement. But yes I would like the `xxxIfPresent` variations to remain, so that we can distinguish between true error, and when a key is not present in a stream. This way we can start to bubble errors up to higher levels. Example use case is `TransactionOptions`, that attempts to decode for any transaction type, so some fields may or may-not be present based on the type of transaction. The current implementation handles decode errors the same as if the field is not present. Which I feel is incorrect... if the key/field is present, but we cannot decode it, that should constitute an error. But if the field is allowed to be optional, then it should be able to return a nil, instead of an error.

This type may also allow for more generic handling of blocks, where a field may or may not be present depending on the chain and fork.

Ideally we need to start laying the path for better error handling, and part of that needs to be able to distinguish between when an error actually happens or not.